thentos-core/src/Thentos/Frontend/State.hs

```diff
@@ -137,21 +127,21 @@ enterFAction aState key smap = Nat $ ExceptT . (>>= _Left fActionServantErr) . r
     run fServer = fst <$> runActionE emptyFrontendSessionData aState fServer'
       where
         fServer' :: FAction a
-        fServer' = do
+        fServer' =
             case smap key of
-                Nothing -> error $ "enterFAction: internal error in servant-session: no cookie!"
+                Nothing ->
+                    -- Should really this be 'error' here.
+                    -- Can a misbehaving client trigger this error? Thus crashing the server?
```

Comment:
Isn't `error` thread-local?  Not sure, but no matter: The error message somewhat obscurely tries to state that servant-session is supposed to always come up with a state, whether freshly created or read from the cookie.  I'll update the comment.  (Please check if this sounds plausible.)